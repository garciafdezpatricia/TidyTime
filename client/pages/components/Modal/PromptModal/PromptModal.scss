@use "../../../../styles/variables" as var;

.backdrop {
    background-color: rgba(0, 0, 0, 0.2);
    width: 100%;
    height: 100%;
    z-index: 9999;
    top: 0;
    left: 0;
    position: absolute;
}

.prompt-modal {
    position: absolute;
    left: 50%;
    right: 50%;
    top: 50%;
    bottom: 50%;
    transform: translate(-50%, -50%);
    width: 40em;
    height: 25em;
    z-index: 9999;
    background-color: var.$white;
    padding: 2em;
    display: flex;
    flex-direction: column;
    border-radius: 5px;
}

.prompt-modal button {
    width: 10em;
    height: 3em;
    display: flex;
    align-items: center;
    justify-content: center;
}

.prompt-modal-button-section {
    display: flex;
    gap: 2em;
    align-self: flex-end;
    width: 100%;
    justify-content: flex-end;
    margin-top: auto;
}

.prompt-modal-secondary-btn {
    background-color: var.$white;
    border: 1px solid var.$black;
    &:hover {
        outline: 1px solid var.$primary-green;
    }
}