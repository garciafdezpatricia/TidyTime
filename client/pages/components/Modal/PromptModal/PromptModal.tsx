import { ReactNode } from "react";
import { Button } from "../../Button/Button";

export interface Props {
    onClose?: () => void;
    onPrimaryAction?: (arg?:any) => void | any;
    onSecondaryAction?: (arg?:any) => void | any;
    primaryActionText?: string;
    secondaryActionText?: string;
    title?: string;
    backdrop: boolean;
    children?: ReactNode;
    variant?: string;
}

export default function PromptModal({
    onClose,
    onPrimaryAction,
    onSecondaryAction,
    primaryActionText,
    secondaryActionText,
    title,
    backdrop,
    children, 
    variant} : Props) {

    return (
        <>
            { backdrop && <div className="backdrop"></div>}
            <article className={variant ? variant : "prompt-modal"}>
                <h1>{title}</h1>
                {children}
                <div className="prompt-modal-button-section">
                    <Button
                        onClick={onSecondaryAction}
                        className="prompt-modal-secondary-btn"
                        >
                        {secondaryActionText}
                    </Button>
                    <Button
                        onClick={onPrimaryAction}
                    >
                        {primaryActionText}
                    </Button>
                </div>
            </article>
        </>
    )
}
