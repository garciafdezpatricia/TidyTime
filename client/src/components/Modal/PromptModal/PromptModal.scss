@use "../../../../styles/variables" as var;

.backdrop {
	background-color: rgba(0, 0, 0, 0.2);
	width: 100%;
	height: 100%;
	z-index: 9;
	top: 0;
	left: 0;
	position: absolute;
}
.prompt-modal,
.rename-list-modal,
.confirmation-modal, .shadow-modal {
	position: absolute;
	transform: translate(-50%, -50%);
	z-index: 9999;
	background-color: var.$white;
	display: flex;
	flex-direction: column;
	border-radius: .4rem;
	box-sizing: border-box;
}
.prompt-modal, .shadow-modal {
	top: 50%;
	left: 50%;
	right: 50%;
	bottom: 50%;
	padding: 1rem 2rem;
	width: 40em;
	height: 25em;
	button {
		width: 5em;
		display: flex;
		justify-content: center;
		padding: .1rem .5rem;
	}
	h3 {
		font-size: 1.2rem;
	}
}

.shadow-modal {
	box-shadow: -.1rem .2rem 1.5rem 0 rgba(0,0,0,0.75);
}

.prompt-modal-button-section {
	display: flex;
	gap: 2em;
	align-self: flex-end;
	width: 100%;
	justify-content: flex-end;
	margin-top: auto;
}
.prompt-modal-secondary-btn {
	background-color: var.$white;
	border: 1px solid var.$black;
	&:hover {
		outline: 1px solid var.$primary-green;
	}
}

.confirmation-modal {
	top: 15%;
	left: 50%;
	right: 50%;
	width: 20rem;
	height: fit-content;
	padding: 1rem;
	border: 1px solid var.$primary-grey;
	align-items: center;
	gap: 1rem;
	p {
		font-size: .8rem;
		text-align: center;
	}
	.primary-button {
		border: none;
		background-color: var.$hover-red;
		color: var.$remove-red;
		&:hover {
			outline: .1rem solid var.$remove-red;
		}
	}
	.prompt-modal-secondary-btn {
		border: none;
		background-color: var.$primary-grey;
		&:hover {
			outline: .1rem solid var.$black;
		}
	}
	button {
		background-color: var.$white;
		width: 5em;
		align-items: center;
		justify-content: center;
		&:hover {
			outline: none;
		}
	}
	.prompt-modal-button-section {
		align-self: center;
		justify-content: center;
		gap: 1rem;
	}
}

.rename-list-modal {
	top: 15%;
	left: 40%;
	padding: 1rem;
	width: 15rem;
	height: 7rem;
	border: solid 1px var.$primary-grey;
	button {
		display: flex;
		padding-inline: 0.5rem;
	}
	input {
		border: solid 1px var.$primary-grey;
		outline: none;
		padding-left: 0.2rem;
		padding-top: 0.1rem;
		padding-bottom: 0.1rem;
		border-radius: 0.4rem;
		&:active,
		&:focus,
		&:hover {
			border: solid 1px var.$primary-grey;
		}
	}
}
