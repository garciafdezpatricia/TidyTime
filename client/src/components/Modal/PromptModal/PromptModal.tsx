import { ReactNode } from "react";
import { Button } from "../../Button/Button";
import { useClickAway } from "@uidotdev/usehooks";

export interface Props {
    onClose?: () => void;
    onPrimaryAction?: (arg?:any) => void | any;
    onSecondaryAction?: (arg?:any) => void | any;
    primaryActionText?: string;
    secondaryActionText?: string;
    title?: string;
    backdrop: boolean;
    children?: ReactNode;
    variant?: string;
}

export default function PromptModal({
    onClose,
    onPrimaryAction,
    onSecondaryAction,
    primaryActionText,
    secondaryActionText,
    title,
    backdrop,
    children, 
    variant} : Props) {

    const ref = useClickAway(() => {
        if (onSecondaryAction) {
            onSecondaryAction()
        }
    })

    return (
        <>
            { backdrop && <div className="backdrop"></div>}
            { 
            // @ts-ignore
            <article ref={ref} className={variant ? variant : "prompt-modal"}>
                <p>{title}</p>
                {children}
                <div className="prompt-modal-button-section">
                    { onSecondaryAction && <Button
                        onClick={onSecondaryAction}
                        className="prompt-modal-secondary-btn"
                        >
                        {secondaryActionText}
                    </Button>}
                    <Button
                        onClick={onPrimaryAction}
                    >
                        {primaryActionText}
                    </Button>
                </div>
            </article>
            }
        </>
    )
}
