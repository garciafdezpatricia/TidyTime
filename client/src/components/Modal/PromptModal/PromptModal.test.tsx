import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import PromptModal from '@/src/components/Modal/PromptModal/PromptModal';

describe('PromptModal without backdrop', () => {
    it('renders without crashing', () => {
      const { container } = render(<PromptModal backdrop={false}/>);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<PromptModal backdrop={false}/>);
        expect(asFragment()).toMatchSnapshot();
      });
  });

  describe('PromptModal with backdrop', () => {
    it('renders without crashing', () => {
      const { container } = render(<PromptModal backdrop={true}/>);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<PromptModal backdrop={true}/>);
        expect(asFragment()).toMatchSnapshot();
      });
  });