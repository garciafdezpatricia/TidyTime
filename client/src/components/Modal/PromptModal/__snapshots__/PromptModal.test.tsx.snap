// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromptModal with backdrop matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="backdrop"
  />
  <article
    class="prompt-modal"
  >
    <p />
    <div
      class="prompt-modal-button-section"
    >
      <button
        class="primary-button"
        data-testid="primary-button"
      />
    </div>
  </article>
</DocumentFragment>
`;

exports[`PromptModal without backdrop matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="prompt-modal"
  >
    <p />
    <div
      class="prompt-modal-button-section"
    >
      <button
        class="primary-button"
        data-testid="primary-button"
      />
    </div>
  </article>
</DocumentFragment>
`;
